cilium-cli/install/helm.go

```diff
@@ -64,9 +64,13 @@ func (k *K8sInstaller) getHelmValues() (map[string]interface{}, error) {
 				// Can be removed once we drop support for <1.14.0
 				helmMapOpts["tunnel"] = tunnelDisabled
 			}
-			// AL2023 uses ens interfaces, but we default to eth interfaces for everything else for backwards compatibility,
-			// since the CLI was assuming eth interfaces before support for AL2023 was introduced
-			helmMapOpts["egressMasqueradeInterfaces"] = "eth+ ens+"
+			if versioncheck.MustCompile(">=1.17.0")(k.chartVersion) {
+				// AL2023 uses ens interfaces, but we default to eth interfaces for everything else for backwards compatibility,
+				// since the CLI was assuming eth interfaces before support for AL2023 was introduced
+				helmMapOpts["egressMasqueradeInterfaces"] = "eth+ ens+"
+			} else {
+				helmMapOpts["egressMasqueradeInterfaces"] = "eth+"
```

Comment:
let's do
```suggestion
				helmMapOpts["egressMasqueradeInterfaces"] = "eth+"
				if k.params.AWS.AwsNodeImageFamily == AwsNodeImageFamilyAmazonLinux2023 {
					helmMapOpts["egressMasqueradeInterfaces"] = "ens+"
				}
```
to retain the previous logic that got modified in https://github.com/cilium/cilium/pull/36504

Reply:
Oh, we don't detect node family anymore: https://github.com/cilium/cilium/pull/36336/files (((

Reply:
I can recover the previous logic but it also not ideal, for example in mixed node clusters.

Reply:
oh i see, then i guess there isn't much we can do here 👍